pkgs/applications/networking/instant-messengers/nheko/default.nix

```diff
@@ -1,70 +1,72 @@
 , nlohmann_json
 }:
 
-# These hashes and revisions are based on those from here:
-# https://github.com/Nheko-Reborn/nheko/blob/v0.6.4/deps/CMakeLists.txt#L52
-let
-  tweeny = fetchFromGitHub {
-    owner = "mobius3";
-    repo = "tweeny";
-    rev = "b94ce07cfb02a0eb8ac8aaf66137dabdaea857cf";
-    sha256 = "1wyyq0j7dhjd6qgvnh3knr70li47hmf5394yznkv9b1indqjx4mi";
-  };
-
-  lmdbxx = fetchFromGitHub {
-    owner = "bendiken";
-    repo = "lmdbxx";
-    rev = "0b43ca87d8cfabba392dfe884eb1edb83874de02";
-    sha256 = "1whsc5cybf9rmgyaj6qjji03fv5jbgcgygp956s3835b9f9cjg1n";
-  };
-in
 mkDerivation rec {
   pname = "nheko";
-  version = "0.6.4";
+  version = "0.7.1";
 
   src = fetchFromGitHub {
     owner = "Nheko-Reborn";
     repo = "nheko";
     rev = "v${version}";
-    sha256 = "19dkc98l1q4070v6mli4ybqn0ip0za607w39hjf0x8rqdxq45iwm";
+    sha256 = "12sxibbrn79sxkf9jrm7jrlj7l5vz15claxrrll7pkv9mv44wady";
   };
 
-  # If, on Darwin, you encounter the error
-  #   error: must specify at least one argument for '...' parameter of variadic
-  #   macro [-Werror,-Wgnu-zero-variadic-macro-arguments]
-  # Then adding this parameter is likely the fix you want.
-  #
-  # However, it looks like either cmake doesn't honor this CFLAGS variable, or
-  # darwin's compiler doesn't have the same syntax as gcc for turning off
-  # -Werror selectively.
-  #
-  # Anyway, this is something that will have to be debugged with access to a
-  # darwin-based OS. Sorry about that!
-  #
-  #preConfigure = lib.optionalString stdenv.isDarwin ''
-  #  export CFLAGS=-Wno-error=gnu-zero-variadic-macro-arguments
-  #'';
-
-  postPatch = ''
-    mkdir -p .deps/include/
-    ln -s ${tweeny}/include .deps/include/tweeny
-    ln -s ${spdlog} .deps/spdlog
-  '';
-
+  nativeBuildInputs = [
+    lmdbxx
+    cmake
+    pkgconfig
+  ];
   cmakeFlags = [
-    "-DTWEENY_INCLUDE_DIR=.deps/include"
-    "-DLMDBXX_INCLUDE_DIR=${lmdbxx}"
-    "-Dnlohmann_json_DIR=${nlohmann_json}/lib/cmake/nlohmann_json"
+    # Can be removed once either https://github.com/NixOS/nixpkgs/pull/85254 or
+    # https://github.com/NixOS/nixpkgs/pull/73940 are merged
+    "-DBoost_NO_BOOST_CMAKE=TRUE"
+  ];
+  # commit missing from latest release and recommended by upstream:
+  # https://github.com/NixOS/nixpkgs/pull/85922#issuecomment-619263903
+  patches = [
+    (fetchpatch {
+      name = "room-ids-escape-patch";
+      url = "https://github.com/Nheko-Reborn/nheko/commit/d94ac86816f9f325cba11f71344a3ca99591130d.patch";
+      sha256 = "1p0kj4a60l3jf0rfakc88adld7ccg3vfjhzia5rf2i03h35cxw8c";
+    })
   ];
-
-  nativeBuildInputs = [ cmake pkgconfig ];
 
   buildInputs = [
-    mtxclient olm boost lmdb spdlog cmark
-    qtbase qtmultimedia qttools
+    nlohmann_json
+    tweeny
+    mtxclient
+    olm
+    boost17x
+    lmdb
+    spdlog
+    cmark
+    qtbase
+    qtmultimedia
+    qttools
+    qtquickcontrols2
+    qtgraphicaleffects
   ] ++ lib.optional stdenv.isDarwin qtmacextras;
 
   enableParallelBuilding = true;
```

Comment:
I'm pretty sure this is redundant.

Reply:
Why?

Reply:
It should be already `true`, at least for Qt applications.

Reply:
That's correct.